### PR TITLE
[windows] Support for Sysmon Registry non-QWORD/DWORD events

### LINKED ISSUE #2864: [windows] Lacks parsing for Sysmon Registry non-QWORD/DWORD events
When using the Windows Integration for collecting Sysmon logs, we found that Sysmon registry logs are not parsed if they are not DWORD or QWORD.

We had to drop sysmon support to various detection rules in https://github.com/elastic/detection-rules/pull/1775 that were ineffective due to this limitation on the integration.

### PR DESCRIPTION
**Pull request by r00tu53r**
## What does this PR do?

- fixes the QWORD and DWORD regex
- enables the `sysmon_operational` pipeline to process sysmon registry events for values other than QWORD and DWORD (`REG_SZ, REG_MULTI_SZ, REG_BINARY, REG_EXPAND_SZ`).

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

`elastic-package test pipeline -v`

## Related issues

- Closes #2864

### CODE REVIEW

File: packages/windows/data_stream/forwarded/sample_event.json [hunk 1]
```diff
@@ -16,17 +17,17 @@
...
+        "version": "7.17.0"
     },
     "event": {
         "agent_id_status": "verified",
         "category": "process",
         "code": "4105",
-        "created": "2022-01-12T05:23:24.033Z",
+        "created": "2022-03-31T08:40:37.999Z",
         "dataset": "windows.forwarded",
-        "ingested": "2022-01-12T05:23:25Z",
```
Comment (efd6) [inline packages/windows/data_stream/forwarded/sample_event.json:29]: Is it worth removing `event.ingested` now?

File: packages/windows/data_stream/sysmon_operational/_dev/test/pipeline/test-events.json-expected.json [hunk 2]
```diff
@@ -21484,6 +21502,504 @@
...
+            "process": {
+                "entity_id": "{ce6f2a55-427a-6245-3801-000000000900}",
+                "executable": "C:\\Windows\\regedit.exe",
+                "name": "regedit.exe",
+                "pid": 5816
+            },
+            "registry": {
+                "data": {
+                    "strings": [
+                        "0x12349abc"
```
Comment (efd6) [inline packages/windows/data_stream/sysmon_operational/_dev/test/pipeline/test-events.json-expected.json:21705]: Why is the DWORD left as hex but the QWORD above is converted to decimal?

File: packages/windows/data_stream/sysmon_operational/elasticsearch/ingest_pipeline/default.yml [hunk 3]
```diff
@@ -1174,34 +1175,55 @@ processors:
...
-              }
+            return;
+          }
+
+          matcher = dwordRegex.matcher(data);
+          if (matcher.matches()) {
+            def parsedValue = Long.parseLong(matcher.group(1).substring(prefixLen), 16);
+            if (!Double.isNaN(parsedValue)) {
+              dataType = "SZ_DWORD";
+              dataValue = matcher.group(1);
```
Comment (efd6) [inline packages/windows/data_stream/sysmon_operational/elasticsearch/ingest_pipeline/default.yml:1203]: Add hex→decimal conversion here.

Comment (r00tu53r) [reply packages/windows/data_stream/sysmon_operational/elasticsearch/ingest_pipeline/default.yml:1203]: I am not sure its the older code. I will convert it to keep it consistent.

**Review by efd6 — APPROVED** [review]

**Comment by r00tu53r** [comment]
@elastic/integrations can someone please help review this PR?

**Review by andrewkroh — APPROVED** [review]
LGTM.

@elastic/integrations, can you please take a look at this.

**Review by jsoriano — APPROVED** [review]

**Pull request merged**